src/pip/_internal/operations/prepare.py

```diff
@@ -503,35 +492,53 @@ def prepare_linked_requirement(self, req, parallel_builds=False):
             return wheel_dist
         return self._prepare_linked_requirement(req, parallel_builds)
 
-    def prepare_linked_requirement_more(self, req, parallel_builds=False):
-        # type: (InstallRequirement, bool) -> None
+    def prepare_linked_requirements_more(self, reqs, parallel_builds=False):
+        # type: (Iterable[InstallRequirement], bool) -> None
         """Prepare a linked requirement more, if needed."""
-        if not req.needs_more_preparation:
-            return
-        self._prepare_linked_requirement(req, parallel_builds)
+        # TODO: Download directly to download_dir instead of
```

Comment:
Similar comment here as with the other TODO, let's break this into a separate issue.